core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -217,10 +223,10 @@ public Dependency(String s) {
             if(idx==-1)
                 throw new IllegalArgumentException("Illegal dependency specifier "+s);
             this.shortName = s.substring(0,idx);
-            this.version = s.substring(idx+1);
-            
+            String version = s.substring(idx+1);
+
             boolean isOptional = false;
-            String[] osgiProperties = s.split(";");
+            String[] osgiProperties = version.split("[;]");
```

Comment:
Not sure what is going on in this hunk. Was the parsing of optional dependencies previously broken somehow?